v3-nginx/docker-entrypoint.d/91-modsecurity-envsubst.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+# vim:sw=2:ts=2:sts=2:et
+
+set -eu
+
+LC_ALL=C
+ME=$( basename "$0" )
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+touch /etc/modsecurity.d/modsecurity-override.conf 2>/dev/null || { echo >&2 "$ME: error: can not modify /etc/modsecurity.d/modsecurity-override.conf (read-only file system?)"; exit 0; }
+
+auto_envsubst() {
+  local template_dir="${MODSEC_ENVSUBST_TEMPLATE_DIR:-/etc/modsecurity.d/templates}"
+  local suffix="${MODSEC_ENVSUBST_TEMPLATE_SUFFIX:-.template}"
+  local output_dir="${MODSEC_ENVSUBST_OUTPUT_DIR:-/etc/modsecurity.d}"
+
+  local template defined_envs relative_path output_path subdir
+  defined_envs=$(printf '${%s} ' $(env | cut -d= -f1))
+  [ -d "$template_dir" ] || return 0
+  if [ ! -w "$output_dir" ]; then
+    echo >&3 "$ME: ERROR: $template_dir exists, but $output_dir is not writable"
+    return 0
+  fi
+  find "$template_dir" -follow -type f -name "*$suffix" -print | while read -r template; do
+    relative_path="${template#$template_dir/}"
+    output_path="$output_dir/${relative_path%$suffix}"
+    subdir=$(dirname "$relative_path")
+    # create a subdirectory where the template file exists
+    mkdir -p "$output_dir/$subdir"
+    echo >&3 "$ME: Running envsubst on $template to $output_path"
```

Comment:
Where are you writing to? Which file is `3`?

Reply:
This came from upstream, but removed in new iteration.